paimon-core/src/test/java/org/apache/paimon/utils/TraceableFileIO.java

```diff
@@ -250,15 +251,13 @@ public String toString() {
     }
 
     public static List<SeekableInputStream> openInputStreams(Predicate<Path> filter) {
-        return OPEN_INPUT_STREAMS.stream()
-                .filter(s -> filter.test(s.file))
-                .collect(Collectors.toList());
+        return new ArrayList<>(OPEN_INPUT_STREAMS)
```

Comment:
It looks like it's also traversed once, is there any difference?

Reply:
copy out to avoid ConcurrentModificationException